keps/sig-node/3815-pod-condition-for-configuration-errors/README.md

```diff
@@ -0,0 +1,1069 @@
+
+It would be ideal to have a standard condition for these common configuration errors.  These errors can be fixed via human interaction but they can sometimes be difficult to programatically fix.  A goal of having a standard condition for these configuration errors would be for third-party controllers and/or Kubernetes controllers to start using a single way of detecting fixable configuration errors.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Add a new condition FailingToStart to detect failures in configuration.
+
+For future cases that we want to add to to this condition, the suggestion would be to consider cases where pods are forever stuck in the pending state.  We want to avoid transient pods that will eventually resolve themselves so we decided to avoid targeting events that are related to ImagePullBackOff because these can eventually be fixed without any interaction.  The general rule could be that pods that require manually intervention to fix could be added as cases for this condition.  
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- Will not modify behavior for ImagePullBackOff
+  - This status can be business as usual or image pull errors
+  - Unclear if it is possible to detect this as a failure
+  - Invalid ImagePullSecrets are a common fail case but it is difficult to distinguish this case from others.
+- Configuration errors that happen during ContainerCreation (Mounting volumes) will not be included as part of this condition
+  - PodReadyToStartContainers condition actually covers these cases so there is no need to cover them in this KEP.
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+- Add a FailingToStart Condition for the following cases:
+
+  - Invalid Image Name
+  - Image not found locally when ImagePullPolicy is set to never.
+  - Existing config map but invalid key reference
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
```

Comment:
Can this section call out how a controller will actually utilize this? Given this condition may usually start out as `True`(as stated as a Risk below), what will be some high-level logic around thresholds in controllers to start taking action based on the condition not becoming `False` vs staying `True` for a pod?

Reply:
Updated.